front/src/views/Boxes/BoxesView.tsx

```diff
@@ -96,8 +96,7 @@ export const ACTION_OPTIONS_FOR_BOXESVIEW_QUERY = gql`
 `;
 
 function Boxes() {
-  const { globalPreferences } = useContext(GlobalPreferencesContext);
-  const baseId = globalPreferences.selectedBase?.id!;
+  const { baseId } = useLoadAndSetGlobalPreferences();
```

Comment:
put baseId retrieval into a separate helper function or hook to make the code more readible.